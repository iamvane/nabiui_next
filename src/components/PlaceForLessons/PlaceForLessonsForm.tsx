import * as React from 'react';

import {
  Checkbox,
  FormGroup,
  FormControlLabel,
  Typography
} from '@material-ui/core';

import SectionTitle from '../common/SectionTitle';
import DistanceSelect from '../common/DistanceSelect';
import { placeForLessonsOptions, PlaceForLessonsComponent } from './constants';
import { PlaceForLessonsType } from './model';

interface Props extends PlaceForLessonsType {
  handleChange: (event: React.FormEvent<{}>) => void;
  isFilter?: boolean;
  [key: string]: any;
}

const PlaceForLessonsForm: React.StatelessComponent<Props> = props => {
  const placeForLessonsItems = [];

  for (const [key, value] of Object.entries(placeForLessonsOptions)) {
    placeForLessonsItems.push(
      <FormControlLabel
        key={key}
        control={
          <Checkbox
            name={value.name}
            checked={props[value.name]}
            onChange={props.handleChange}
          />
        }
        label={value.label}
      />
    );
  }

  const distanceField = props.home && !props.isFilter && (
    <DistanceSelect
      handleChange={props.handleChange}
      distance={props.distance}
    />
  );

  return (
    <div>
      <SectionTitle text={PlaceForLessonsComponent.sectionTitle} />
      {!props.isFilter && <Typography className="nabi-margin-top-xsmall nabi-margin-bottom-xsmall">
        {PlaceForLessonsComponent.description}
      </Typography>
      }
      <FormGroup className="nabi-margin-left-small">
        {placeForLessonsItems}
      </FormGroup>
    </div>
  );
};

export default (PlaceForLessonsForm);
