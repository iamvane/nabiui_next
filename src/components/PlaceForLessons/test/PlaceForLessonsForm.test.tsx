import * as React from 'react';
import PlaceForLessonsForm from 'components/PlaceForLessons/PlaceForLessonsForm';
import { shallow } from 'enzyme';

describe('PlaceForLessonsForm', () => {
  let wrapper: any;

  const handleChange: () => void = jest.fn();
  const handleChangePlaceForLessons: () => void = jest.fn();

  describe('Shallow', () => {
    beforeEach(() => {
      wrapper = shallow(
        <PlaceForLessonsForm
          handleChange={handleChange}
          handleChangePlaceForLessons={handleChangePlaceForLessons}
          home={false}
          studio={false}
          online={false}
          studioAddress=""
        />
      );
    });

    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
