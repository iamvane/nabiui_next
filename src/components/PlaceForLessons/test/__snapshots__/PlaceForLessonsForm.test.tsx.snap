// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceForLessonsForm Shallow Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlaceForLessonsForm
    handleChange={[MockFunction]}
    handleChangePlaceForLessons={[MockFunction]}
    home={false}
    online={false}
    studio={false}
    studioAddress=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SectionTitle
          text="Place for lessons"
        />,
        <WithStyles(Typography)
          className="nabi-margin-top-xsmall nabi-margin-bottom-xsmall"
        >
          Specify the place(s) where you are available to teach (select all that apply)
        </WithStyles(Typography)>,
        <WithStyles(FormGroup)
          className="nabi-margin-left-small"
        >
          <WithStyles(WithFormControlContext(FormControlLabel))
            control={
              <WithStyles(Checkbox)
                checked={false}
                name="home"
                onChange={[MockFunction]}
              />
            }
            label="Student's home"
          />
          <WithStyles(WithFormControlContext(FormControlLabel))
            control={
              <WithStyles(Checkbox)
                checked={false}
                name="studio"
                onChange={[MockFunction]}
              />
            }
            label="Instructor's studio"
          />
          <WithStyles(WithFormControlContext(FormControlLabel))
            control={
              <WithStyles(Checkbox)
                checked={false}
                name="online"
                onChange={[MockFunction]}
              />
            }
            label="Online"
          />
        </WithStyles(FormGroup)>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "text": "Place for lessons",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Specify the place(s) where you are available to teach (select all that apply)",
          "className": "nabi-margin-top-xsmall nabi-margin-bottom-xsmall",
        },
        "ref": null,
        "rendered": "Specify the place(s) where you are available to teach (select all that apply)",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="home"
                  onChange={[MockFunction]}
                />
              }
              label="Student's home"
            />,
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="studio"
                  onChange={[MockFunction]}
                />
              }
              label="Instructor's studio"
            />,
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="online"
                  onChange={[MockFunction]}
                />
              }
              label="Online"
            />,
          ],
          "className": "nabi-margin-left-small",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Home",
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Checkbox)
                checked={false}
                name="home"
                onChange={[MockFunction]}
              />,
              "label": "Student's home",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Studio",
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Checkbox)
                checked={false}
                name="studio"
                onChange={[MockFunction]}
              />,
              "label": "Instructor's studio",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Online",
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Checkbox)
                checked={false}
                name="online"
                onChange={[MockFunction]}
              />,
              "label": "Online",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SectionTitle
            text="Place for lessons"
          />,
          <WithStyles(Typography)
            className="nabi-margin-top-xsmall nabi-margin-bottom-xsmall"
          >
            Specify the place(s) where you are available to teach (select all that apply)
          </WithStyles(Typography)>,
          <WithStyles(FormGroup)
            className="nabi-margin-left-small"
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="home"
                  onChange={[MockFunction]}
                />
              }
              label="Student's home"
            />
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="studio"
                  onChange={[MockFunction]}
                />
              }
              label="Instructor's studio"
            />
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  name="online"
                  onChange={[MockFunction]}
                />
              }
              label="Online"
            />
          </WithStyles(FormGroup)>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "text": "Place for lessons",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Specify the place(s) where you are available to teach (select all that apply)",
            "className": "nabi-margin-top-xsmall nabi-margin-bottom-xsmall",
          },
          "ref": null,
          "rendered": "Specify the place(s) where you are available to teach (select all that apply)",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(WithFormControlContext(FormControlLabel))
                control={
                  <WithStyles(Checkbox)
                    checked={false}
                    name="home"
                    onChange={[MockFunction]}
                  />
                }
                label="Student's home"
              />,
              <WithStyles(WithFormControlContext(FormControlLabel))
                control={
                  <WithStyles(Checkbox)
                    checked={false}
                    name="studio"
                    onChange={[MockFunction]}
                  />
                }
                label="Instructor's studio"
              />,
              <WithStyles(WithFormControlContext(FormControlLabel))
                control={
                  <WithStyles(Checkbox)
                    checked={false}
                    name="online"
                    onChange={[MockFunction]}
                  />
                }
                label="Online"
              />,
            ],
            "className": "nabi-margin-left-small",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Home",
              "nodeType": "class",
              "props": Object {
                "control": <WithStyles(Checkbox)
                  checked={false}
                  name="home"
                  onChange={[MockFunction]}
                />,
                "label": "Student's home",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Studio",
              "nodeType": "class",
              "props": Object {
                "control": <WithStyles(Checkbox)
                  checked={false}
                  name="studio"
                  onChange={[MockFunction]}
                />,
                "label": "Instructor's studio",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Online",
              "nodeType": "class",
              "props": Object {
                "control": <WithStyles(Checkbox)
                  checked={false}
                  name="online"
                  onChange={[MockFunction]}
                />,
                "label": "Online",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
